python/pyspark/pandas/namespace.py

```diff
@@ -1186,9 +1188,29 @@ def read_excel_on_spark(
                 pdf = pdf_or_pser
 
             psdf = cast(DataFrame, from_pandas(pdf))
-            return_schema = force_decimal_precision_scale(
-                as_nullable_spark_type(psdf._internal.spark_frame.drop(*HIDDEN_COLUMNS).schema)
```

Comment:
`as_nullable_spark_type` convert all fields (both index and data) to nullable=true, while in `InternalFrame` it assert the index field should have nullable=false